pkg/endpoint/endpoint.go

```diff
@@ -1829,7 +1833,12 @@ func (e *Endpoint) setPolicyRevision(rev uint64) {
 			ps.done(now)
 			delete(e.policyRevisionSignals, ps)
 		default:
-			if rev >= ps.wantedRev {
+			// The policy revision of an endpoint can be bumped
+			// before the initial build has happened if an endpoint
+			// is not affected by a policy update. Require an
+			// initial build to have happened to assume that a
+			// policy revision is implemented.
```

Comment:
Nit: I think the comment above `WaitForPolicyRevision` needs to be updated as well:
https://github.com/cilium/cilium/blob/a0fcbd73529f456fd49c85419474a580a9a7fa11/pkg/endpoint/endpoint.go#L1864-L1869